presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
I looked into #7354 and I'm using `unmodifiableSet(map.keySet())` there. If it's ok I'm using the non-linked version, then I'm ok with just throwing in every currently-broken method in ILHM.